### ISSUE TITLE
docs: advertise composition of polynomials

### ISSUE THREAD

**Issue by nsajko** [JuliaMath/Polynomials.jl#509]
Searching the docs for "composition" or "substitution" doesn't turn up anything. The same code path designed for evaluating a polynomial seems to work fine for composing two polynomials together, though:
```julia-repl
julia> using Polynomials

julia> p = Polynomial([0, 1])
Polynomial(x)

julia> q = Polynomial(Float64[0, 0.5])
Polynomial(0.5*x)

julia> r = Polynomial(Float64[0, 2])
Polynomial(2.0*x)

julia> q(r) == r(q) == p == p(p) == p(p(p)) == p(p(p(p)))
```

**Comment by jverzani** [comment]
Yes, one means of converting from one polynomial type to another is along the lines of

`convert(P, q::Q) = q(variable(P))` (P,Q some polynomial type)

But perhaps (given your other issue) we should have a special implementation for (p::P)(q::Q) beyond `evalpoly`.
